### PR TITLE
LPS-84509 configure the generateSampleProject task to include correct arguments

### PR DESCRIPTION
**Pull request by gamerson**

### CODE REVIEW

**Comment by gamerson** [comment]
ci:test:sf

**Comment by gamerson** [comment]
ci:test:project-templates

File: modules/build.gradle [hunk 1]
```diff
@@ -106,6 +106,13 @@ gradle.beforeProject {
 				enabled = false
 			}
 
+			task generateSampleProject(type:JavaExec) {
```
Comment (petershin) [inline modules/build.gradle]: Split out task creation and move it to the top when possible.

```gradle
if (project.path.startsWith(":sdk:project-templates:project-templates-")) {
	task generateSampleProject(type:JavaExec)

	buildCSS {
		enabled = false
	}

	generateSampleProject {
		...
	}
```

File: modules/build.gradle [hunk 2]
```diff
@@ -106,6 +106,13 @@ gradle.beforeProject {
...
+				args '--archetypes-dirs', jar.archivePath.parent, '--destination', "${buildDir}", '--template', project.name.replaceFirst("project-templates-", "")
```
Comment (petershin) [inline modules/build.gradle]: Always use double quotes, unless single quotes are necessary.

File: modules/build.gradle [hunk 3]
```diff
@@ -106,6 +106,13 @@ gradle.beforeProject {
...
+				classpath = sourceSets.main.runtimeClasspath
+				dependsOn jar
+				main = 'com.liferay.project.templates.ProjectTemplates'
```
Comment (petershin) [inline modules/build.gradle]: Always use double quotes, unless single quotes are necessary.

File: modules/sdk/project-templates/project-templates-social-bookmark/build.gradle [hunk 4]
```diff
@@ -1,4 +1,15 @@
...
-	compileOnly group: "com.liferay", name: "com.liferay.project.templates", version: "4.1.6"
+	compile project(":sdk:project-templates:project-templates")
+
 	compileOnly group: "org.apache.maven.archetype", name: "archetype-common", version: "2.4"
+}
+
+generateSampleProject {
+	String sampleProjectName = "foo"
+
+	args '--class-name', 'Foo', '--liferayVersion', '7.1', '--name', sampleProjectName
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-social-bookmark/build.gradle]: Always use double quotes, unless single quotes are necessary.

File: modules/sdk/project-templates/project-templates-social-bookmark/build.gradle [hunk 5]
```diff
@@ -1,4 +1,15 @@
 dependencies {
-	compileOnly group: "com.liferay", name: "com.liferay.project.templates", version: "4.1.6"
+	compile project(":sdk:project-templates:project-templates")
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-social-bookmark/build.gradle]: If it doesn't break any functionality, please apply this change to all the other `project-template` subprojects for consistency.

File: modules/sdk/project-templates/project-templates-social-bookmark/build.gradle [hunk 6]
```diff
@@ -1,4 +1,15 @@
...
+
 	compileOnly group: "org.apache.maven.archetype", name: "archetype-common", version: "2.4"
+}
+
+generateSampleProject {
+	String sampleProjectName = "foo"
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates-social-bookmark/build.gradle]: Please remove `sampleProjectName` and hardcode it to reduce the number of lines in build.gradle

File: modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/util/FileUtil.java [hunk 7]
```diff
@@ -211,6 +211,30 @@ public static Path getFile(Path dirPath, String glob) throws IOException {
...
+
+		try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(
+				dirPath, glob)) {
+
+			Iterator<Path> iterator = directoryStream.iterator();
+
+			while (iterator.hasNext()) {
+				Path path = iterator.next();
+
+				Path fileName = path.getFileName();
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/util/FileUtil.java]: The variable name should end with the class name for readability (i.e. `fileNamePath`).

File: modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/util/FileUtil.java [hunk 8]
```diff
@@ -211,6 +211,30 @@ public static Path getFile(Path dirPath, String glob) throws IOException {
...
+
+				String fileNameMatch = fileName.toString();
```
Comment (petershin) [inline modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/util/FileUtil.java]: The variable `fileNameMatch` can be renamed to `fileName`.

**Comment by gamerson** [comment]
@petershin fixed!

**Comment by petershin** [comment]
Sent to Brian: https://github.com/brianchandotcom/liferay-portal/pull/62812

**Pull request closed without merging**